README.md

```diff
@@ -1,19 +1,111 @@
+   on your API, there are expectations around quality, uptime, response
+time.
+
+You also have to consider things like **rate limiting**, where you
+monitor access and only allow a certain number of API calls per
+application in a given timeframe.
+
+We won't be covering much of this in this unit, but these are all
+important considerations if you plan to make your API public.
+
+### Internal API Considerations
+
+An internal API is a great way to provide access to data for multiple internal
+applications. While there are still many things to consider, the
+concerns for internal APIs are generally less strenuous because you are
+usually in control of all of the applications that will have access.
+
+**Documentation:** Documentation is still important, but with an
+   internal API you generally won't have to provide support for such a
+wide array of consumers, and the docs can be a little more utilitarian.
+
+**Versioning:** Versioning is still a consideration. Depending on
+   your development process, it's possible that not all of your apps
+will be able to respond to changes at the same time, so you may still
+need versioning to support that. However, the consumers will all be
+known, so you won't have to worry as much about some rogue developer
+steadfastly holding on to the beta version of your API.
+
+**Authentication and Security:** Security is still a concern, of
+   course, but again, you'll be in control of what applications can
+access your API.
+
+We don't want to store access keys or passwords on the client of your
+internal API in cookies or JavaScript, because then it wouldn't be secure. And depending on the
+use of the API, it might not make sense to do user-based authentication
+at all. So how do we ensure that only valid web applications are
+accessing our internal API?
+
+Browsers implement something called [Same Origin Security Policy](https://en.wikipedia.org/wiki/Same-origin_policy), which prevents a page from running a script to access data from a different origin.
+
+This is great because by default it means your API can't be accessed by
+another domain. But what about the ones you want to give access to? For
+that, we can configure [Cross Origin Resource Sharing](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing), or **CORS**, to allow whitelist access to our domains while still blocking others. We won't be doing much with this in this unit, but there are great tools like [rack-cors](https://github.com/cyu/rack-cors) out there to help.
+
+**Quality of Service:** You don't get to slack on this one with your
+internal APIs. In fact, there's a good chance that your internal API is
+mission-critical and handles more requests than an external API might.
+
+You might not need to worry about rate limiting, but peformance, uptime,
+and response time are still extremely important no matter who's
+accessing your API.
+
+## Summary
+
+We've looked at the reasons you might choose to create an internal or
```

Comment:
This summary is really helpful. The tone of this lesson is great, it just gets a little dense in the section about security (understandably). Simplifying that section will really tighten the lesson up and help it all cohere.